projects/smart-table/src/lib/components/smart-table/smart-table.component.ts

```diff
@@ -173,113 +162,39 @@ export class SmartTableComponent implements OnInit, OnDestroy {
-        filter(config => !!config),
-        switchMap((customConfig) => combineLatest(of(customConfig), this.configuration$).pipe(first())),
-        map(([customConfig, configuration]) => {
-          // Whenever we have custom configuration coming in, override existing configuration
-          return {
-            ...configuration,
-            ...customConfig,
-            options: {
-              ...configuration.options,
-              ...customConfig.options
-            }
-          };
-        })
-      )
-    ).pipe(
-      // Every time configuration changes, get the columns from the storage again
-      // (because storage identifier will most likely have changed)
-      map(config => config.options.persistTableConfig ? this.getLocalStorageObject(config) : config),
-      shareReplay(1)
-    );
+    // First setup the 3 sources where configuration may come from:
+    // backend, custom configuration via @input and stored configuration
+    this.configurationService.initConfiguration({
+      id: this.instanceId,
+      backendCallback: () => this.dataService.getConfiguration(this.apiUrl, this.httpHeaders),
+      customConfiguration$: this.customConfiguration$,
+      storageCallback: (config) => of(this.storageService.getConfiguration(config)),
+    });
+    this.configuration$ = this.configurationService.getConfiguration(this.instanceId);
 
-    // Columns are extracted from configuration
-    this.allColumns$ = this.configuration$.pipe(
-      tap((config) => this.resetOrderBy(config && config.options && config.options.defaultSortOrder)),
-      map((config: SmartTableConfig) =>
-        config.columns.map(c =>
-          this.factory.createTableColumnFromConfig(c, this.columnTypes, SMARTTABLE_DEFAULT_OPTIONS))),
-      startWith([]),
-      shareReplay(1)
-    );
 
-    // Persist columns configuration in storage whenever we show/hide columns
-    this.persistInStorage$ = this.toggleHideColumn$.pipe(
-      tap(() => this.flyOutService.close()),
-      switchMap(() => combineLatest(this.configuration$, this.selectableColumns$)),
-      filter(([config, selectableColumns]: [SmartTableConfig, TableColumn[]]) =>
-        config.options.persistTableConfig === true && !!config.options.storageIdentifier && !!selectableColumns),
-      map(([configuration, selectableColumns]) => {
-        configuration.columns.map(column => {
-          const found = (selectableColumns as Array<TableColumn>).find(c => c.value === column.key);
-          if (found) {
-            column.visible = !found.hidden;
-          }
-          return column;
-        });
-        return configuration;
-      }),
-      tap((config: SmartTableConfig) =>
-        this.addToLocalStorage(config.options.storageIdentifier, 'columns', config.columns)),
-      map(config => config.columns)
-    );
+    // Set up persistence hook
+    this.configuration$.pipe(
+      takeUntil(this.destroy$),
+      distinctUntilChanged(),
+      skip(2),
```

Comment:
moet dit niet skip(1) zijn? Ik dacht altijd dat de eerste de oude value was en de de de nieuwe?

Reply:
De eerste value is de backend config, de tweede de backend + default options, vanaf de derde krijgen we de volledige config binnen